src/keysym.c

```diff
@@ -122,6 +128,98 @@ xkb_keysym_is_assigned(xkb_keysym_t ks)
            find_keysym_index(ks) != -1;
 }
 
+struct xkb_keysym_iterator {
+    bool explicit;       /* If true, traverse only explicitly named keysyms */
+    int32_t index;       /* Current index in `keysym_to_name` */
+    xkb_keysym_t keysym; /* Current keysym */
+};
+
+struct xkb_keysym_iterator*
+xkb_keysym_iterator_new(bool explicit)
```

Comment:
How about `iterate_only_explicit_syms`? It's only ever typed here once anyway so having an awkwardly long name means it self-documents :)